streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -512,4 +630,16 @@ public String newName(final String prefix) {
         return prefix + String.format("%010d", index.getAndIncrement());
     }
 
+    /**
+     * <strong>This function is only for internal usage only and should not be called.</strong>
+     * <p>
+     * Create a unique state store name.
+     *
+     * @param prefix processor name prefix
+     * @return a new unique name
+     */
+    public String newStoreName(final String prefix) {
```

Comment:
Can't we just move this into a helper class in package `internal` -- if it's public there is does not matter. As we don't get KIP-120 for `0.11` it would be a step backwards and we would need to update KIP-120, too.